pilot/pkg/security/model/authentication_test.go

```diff
@@ -87,81 +147,88 @@ func TestConstructSdsSecretConfig(t *testing.T) {
 			sdsUdsPath:     "/tmp/sdsuds.sock",
 			expected:       nil,
 		},
 		{
+			name:           "ConstructSdsSecretConfig without serviceAccount",
 			serviceAccount: "",
 			sdsUdsPath:     "spiffe://cluster.local/ns/bar/sa/foo",
 			expected:       nil,
 		},
 	}
 
 	for _, c := range cases {
-		if got := ConstructSdsSecretConfig(c.serviceAccount, c.sdsUdsPath); !reflect.DeepEqual(got, c.expected) {
-			t.Errorf("ConstructSdsSecretConfig: got(%#v) != want(%#v)\n", got, c.expected)
-			fmt.Println(got)
-			fmt.Println(c.expected)
-		}
+		t.Run(c.name, func(t *testing.T) {
+			if got := ConstructSdsSecretConfig(c.serviceAccount, c.sdsUdsPath); !cmp.Equal(got, c.expected, protocmp.Transform()) {
+				t.Errorf("ConstructSdsSecretConfig: got(%#v), want(%#v)\n", got, c.expected)
+			}
+		})
 	}
 }
 
-func TestConstructSdsSecretConfigWithCustomUds(t *testing.T) {
-	cases := []struct {
-		serviceAccount string
-		sdsUdsPath     string
-		expected       *auth.SdsSecretConfig
+func TestConstructValidationContext(t *testing.T) {
+	testCases := []struct {
+		name            string
+		rootCAFilePath  string
+		subjectAltNames []string
+		expected        *auth.CommonTlsContext_ValidationContext
 	}{
 		{
-			serviceAccount: "spiffe://cluster.local/ns/bar/sa/foo",
-			sdsUdsPath:     "/tmp/sdsuds.sock",
-			expected: &auth.SdsSecretConfig{
-				Name: "spiffe://cluster.local/ns/bar/sa/foo",
-				SdsConfig: &core.ConfigSource{
-					InitialFetchTimeout: features.InitialFetchTimeout,
-					ConfigSourceSpecifier: &core.ConfigSource_ApiConfigSource{
-						ApiConfigSource: &core.ApiConfigSource{
-							ApiType: core.ApiConfigSource_GRPC,
-							GrpcServices: []*core.GrpcService{
-								{
-									TargetSpecifier: &core.GrpcService_GoogleGrpc_{
-										GoogleGrpc: &core.GrpcService_GoogleGrpc{
-											TargetUri:  "/tmp/sdsuds.sock",
-											StatPrefix: SDSStatPrefix,
-										},
-									},
-								},
+			name:            "default_ca",
+			rootCAFilePath:  "/root/ca",
+			subjectAltNames: []string{"SystemCACertificates.keychain", "SystemRootCertificates.keychain"},
+			expected: &auth.CommonTlsContext_ValidationContext{
+				ValidationContext: &auth.CertificateValidationContext{
+					TrustedCa: &core.DataSource{
+						Specifier: &core.DataSource_Filename{
+							Filename: "/root/ca",
+						},
+					},
+					MatchSubjectAltNames: []*matcher.StringMatcher{
+						{
+							MatchPattern: &matcher.StringMatcher_Exact{
+								Exact: "SystemCACertificates.keychain",
+							},
+						},
+						{
+							MatchPattern: &matcher.StringMatcher_Exact{
+								Exact: "SystemRootCertificates.keychain",
 							},
-							RefreshDelay: nil,
 						},
 					},
 				},
 			},
 		},
 		{
-			serviceAccount: "",
-			sdsUdsPath:     "/tmp/sdsuds.sock",
-			expected:       nil,
-		},
-		{
-			serviceAccount: "spiffe://cluster.local/ns/bar/sa/foo",
-			sdsUdsPath:     "",
-			expected:       nil,
+			name:           "default_ca no subjectAltNames",
```

Comment:
it's a strange to say `default_ca`, just `default CA without subjectAltNames`. also for the `default_ca` above.